konan_sdk/endpoints/base_endpoint.py

```diff
@@ -0,0 +1,97 @@
+import json
+import requests
+from loguru import logger
+from typing import Dict, Union
+from abc import abstractmethod
+
+
+class KonanBaseEndpoint:
+    """
+    Base Endpoint class for Konan endpoints to inherit.
+    """
+
+    name = ''  # endpoint name to reference in logs
+
+    def __init__(self, api_url: str, user, **kwargs) -> None:
```

Comment:
I think it would help if user was marked as nullable cause there are some cases which it can be ( e.g LoginEndpoint)